### PR TITLE
Upgrade `get-stream`

### PR DESCRIPTION
**Pull request by ehmicky**
This upgrades `get-stream` to its latest version.

### CODE REVIEW

File: test/stream.js [hunk 1]
```diff
@@ -17,6 +17,25 @@ test('buffer', async t => {
 	t.is(stdout.toString(), 'foo');
 });
 
+const checkEncoding = async (t, encoding) => {
+	const encodedString = '\u1000.';
```
Comment (ehmicky) [inline test/stream.js:21]: The string was chosen to give different outputs with each encoding type.

File: lib/stream.js [hunk 2]
```diff
@@ -93,11 +97,21 @@ const getStreamPromise = (stream, {encoding, buffer, maxBuffer}) => {
 		return;
 	}
 
-	if (encoding) {
-		return getStream(stream, {encoding, maxBuffer});
+	// eslint-disable-next-line unicorn/text-encoding-identifier-case
+	if (encoding === 'utf8' || encoding === 'utf-8') {
```
Comment (ehmicky) [inline lib/stream.js:101]: Allows both `utf8` and `utf-8` to mimic Node.js behavior.

File: lib/stream.js [hunk 3]
```diff
@@ -93,11 +97,21 @@ const getStreamPromise = (stream, {encoding, buffer, maxBuffer}) => {
...
+		return getStreamAsBuffer(stream, {maxBuffer});
 	}
 
-	return getStream.buffer(stream, {maxBuffer});
+	return applyEncoding(stream, maxBuffer, encoding);
+};
+
+const applyEncoding = async (stream, maxBuffer, encoding) => {
+	const buffer = await getStreamAsBuffer(stream, {maxBuffer});
+	return buffer.toString(encoding);
```
Comment (ehmicky) [inline lib/stream.js:114]: There is one subtle change of behavior here (that is quite of an edge case): when using both `maxBuffer` and an unusual `encoding` (that is neither `utf8` nor `null`, such as `hex` or `base64`), then the number of bytes being limited by `maxBuffer` is stdout/stderr as a `Buffer`, not as the final value of `childProcess.stdout/stderr`, which is an hex/base64/etc.-encoded string.

File: test/stream.js [hunk 4]
```diff
@@ -17,6 +17,25 @@ test('buffer', async t => {
...
+test('can pass encoding "utf16le"', checkEncoding, 'utf16le');
+test('can pass encoding "utf-16le"', checkEncoding, 'utf16le');
+test('can pass encoding "ucs2"', checkEncoding, 'utf16le');
+test('can pass encoding "ucs-2"', checkEncoding, 'utf16le');
+test('can pass encoding "latin1"', checkEncoding, 'latin1');
+test('can pass encoding "binary"', checkEncoding, 'latin1');
+test('can pass encoding "ascii"', checkEncoding, 'ascii');
+test('can pass encoding "hex"', checkEncoding, 'hex');
+test('can pass encoding "base64"', checkEncoding, 'base64');
+test('can pass encoding "base64url"', checkEncoding, 'base64url');
```
Comment (ehmicky) [inline test/stream.js:37]: Ensure we support all the `encoding` option's value that `childProcess.exec()` supports.

File: test/stream.js (hunk 1 above)
Comment (sindresorhus) [reply test/stream.js:21]: I would write that as a short code comment as it's definitely not clear when just reading the code.

Comment (ehmicky) [reply test/stream.js]: Done in cff527d6698016392dbc92a2cac6832e82a5fb66 :+1:

**Comment by sindresorhus** [comment]
Yeah, go ahead 👍

**Comment by ehmicky** [comment]
Done in [`8.0.0`](https://github.com/sindresorhus/execa/releases/tag/v8.0.0).

**Pull request merged**